pallets/did/src/lib.rs

```diff
@@ -292,6 +344,92 @@ impl DidDetails {
 	}
 }
 
+/// Generates a new DID entry starting from the current one stored in the
+/// storage and by applying the changes in the DIDUpdateOperation. The operation
+/// fails with a DidError if the update operation instructs to delete a
+/// verification key that is not associated with the DID or if the operation
+/// counter is not larger than the one stored on chain. !!! To note that this
+/// method does not perform any checks regarding the validity of the
+/// DIDUpdateOperation signature.
+impl<DidIdentifier> TryFrom<(DidDetails, DidUpdateOperation<DidIdentifier>)> for DidDetails
+where
+	DidIdentifier: Parameter + Encode + Decode + Debug,
+{
+	type Error = DidError;
+
+	fn try_from(value: (DidDetails, DidUpdateOperation<DidIdentifier>)) -> Result<Self, Self::Error> {
+		let (old_details, update_operation) = value;
+		// Old attestation key is used later in the process, so it's saved here.
+		let old_attestation_key = old_details.attestation_key;
+		// Copy old state into new, and apply changes in operation to new state.
+		let mut new_details = old_details;
+
+		// Start from all the current verification keys...
+		let mut remaining_verification_keys = new_details.verification_keys;
+
+		// If there are some keys to remove...
+		if let Some(verification_keys_to_remove) = update_operation.verification_keys_to_remove {
+			// Keep track of the keys already seen so far (as we are using a list instead of
+			// a set for the keys)
+			let mut seen_keys = BTreeSet::<PublicVerificationKey>::new();
+
+			verification_keys_to_remove.iter().try_for_each(|key| {
```

Comment:
I would also prefer to use the `BTreeSet`! So I guess I'll just revert back to that version, if you also agree that is a better way to go.

Regarding the `for .. in` stuff, I used the `try_for_each` to allow for exit returns in case of errors. Is that also possible with the classical `foreach` construct?

Reply:
Yes that's possible. It's a bit overloaded. [If you think of an if-block that returns a value](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=04931d20ad8f0c641ae893131f7b1c54):

The if block returns something into x and can also bail early and return something for the whole function. That's a bit strange.

Reply:
Reverting back to sets solves the issue altogether since we use set operations 👍

Reply:
> Yes that's possible. It's a bit overloaded. [If you think of an if-block that returns a value](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=04931d20ad8f0c641ae893131f7b1c54):
> 
> The if block returns something into x and can also bail early and return something for the whole function. That's a bit strange.
> 
> Also: [exit on error in a for loop](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e20888ea698133435acfd54a327b4b07).

I was dealing with a similar issue recently and was also wondering whether `for ... in` was the way to go. I tried to get `map` working but failed.